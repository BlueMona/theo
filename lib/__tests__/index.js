// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

const path = require('path')

const theo = require('../')

const getFixture = file =>
  path.resolve(__dirname, '../__fixtures__/', file)

describe('transform', () => {
  it('works', done => {
    theo
      .transform({
        type: 'raw',
        file: getFixture('a.json')
      })
      .fold(
        e => { throw new Error(e) },
        r => {
          expect(r.delete('meta')).toMatchSnapshot()
          done()
        }
      )
  })
})

describe('format', () => {
  it('works', done => {
    theo
      .transform({
        type: 'raw',
        file: getFixture('a.json')
      })
      .chain(data =>
        theo.format({
          type: 'raw.json',
          data
        })
      )
      .fold(
        e => { throw new Error(e) },
        r => {
          expect(r).toMatchSnapshot()
          done()
        }
      )
  })
})

describe('convertSync', () => {
  it('works', () => {
    const r = theo.convertSync({
      transform: {
        type: 'raw',
        file: getFixture('a.json')
      },
      format: {
        type: 'raw.json'
      }
    })
    expect(r).toMatchSnapshot()
  })
})

describe('convert', () => {
  it('works', done => {
    theo.convert({
      transform: {
        type: 'raw',
        file: getFixture('a.json')
      },
      format: {
        type: 'raw.json'
      }
    })
    .then(r => {
      expect(r).toMatchSnapshot()
      done()
    })
  })
})
