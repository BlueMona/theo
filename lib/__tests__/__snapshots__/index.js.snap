// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert works 1`] = `
"{
  \\"aliases\\": {},
  \\"props\\": {
    \\"TOKEN_B\\": {
      \\"category\\": \\"test\\",
      \\"type\\": \\"test\\",
      \\"value\\": \\"b\\",
      \\"name\\": \\"TOKEN_B\\"
    },
    \\"TOKEN_A\\": {
      \\"category\\": \\"test\\",
      \\"type\\": \\"test\\",
      \\"value\\": \\"a\\",
      \\"name\\": \\"TOKEN_A\\"
    }
  }
}"
`;

exports[`convertSync works 1`] = `
"{
  \\"aliases\\": {},
  \\"props\\": {
    \\"TOKEN_B\\": {
      \\"category\\": \\"test\\",
      \\"type\\": \\"test\\",
      \\"value\\": \\"b\\",
      \\"name\\": \\"TOKEN_B\\"
    },
    \\"TOKEN_A\\": {
      \\"category\\": \\"test\\",
      \\"type\\": \\"test\\",
      \\"value\\": \\"a\\",
      \\"name\\": \\"TOKEN_A\\"
    }
  }
}"
`;

exports[`format works 1`] = `
"{
  \\"aliases\\": {},
  \\"props\\": {
    \\"TOKEN_B\\": {
      \\"category\\": \\"test\\",
      \\"type\\": \\"test\\",
      \\"value\\": \\"b\\",
      \\"name\\": \\"TOKEN_B\\"
    },
    \\"TOKEN_A\\": {
      \\"category\\": \\"test\\",
      \\"type\\": \\"test\\",
      \\"value\\": \\"a\\",
      \\"name\\": \\"TOKEN_A\\"
    }
  }
}"
`;

exports[`transform works 1`] = `
Immutable.Map {
  aliases: Immutable.Map {
  },
  props: Immutable.Map {
    TOKEN_B: Immutable.Map {
      category: "test",
      type: "test",
      value: "b",
      name: "TOKEN_B",
    },
    TOKEN_A: Immutable.Map {
      category: "test",
      type: "test",
      value: "a",
      name: "TOKEN_A",
    },
  },
}
`;
